samples/msal-node-samples/on-behalf-of-distributed-cache/src/AuthProvider.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { performance } from "perf_hooks";
+import { RedisClientType } from "redis";
+import {
+    Configuration,
+    LogLevel,
+    ConfidentialClientApplication,
+    AuthenticationResult,
+    CryptoProvider,
+    OnBehalfOfRequest
+} from "@azure/msal-node";
+
+import CustomCachePlugin from "./CustomCachePlugin";
+import RedisClientWrapper from "./RedisClientWrapper";
+import AxiosHelper from './AxiosHelper';
+
+export type AppConfig = {
+    instance: string;
+    tenantId: string;
+    clientId: string;
+    clientSecret: string;
+    permissions: string;
+};
+
+export class AuthProvider {
+    private msalConfig: Configuration;
+    private cryptoProvider: CryptoProvider;
+    private cacheClientWrapper: RedisClientWrapper;
+
+    private constructor(msalConfig: Configuration, cacheClient: RedisClientType) {
+        this.msalConfig = msalConfig;
+        this.cryptoProvider = new CryptoProvider();
+        this.cacheClientWrapper = new RedisClientWrapper(cacheClient);
+    }
+
+    static async initialize(appConfig: AppConfig, cacheClient: RedisClientType): Promise<AuthProvider> {
+        const msalConfig = {
+            auth: {
+                clientId: appConfig.clientId,
+                authority: `${appConfig.instance}/${appConfig.tenantId}`,
+                clientSecret: appConfig.clientSecret,
+            },
+            system: {
+                loggerOptions: {
+                    loggerCallback(loglevel, message, containsPii) {
+                        console.log(message);
+                    },
+                    logLevel: LogLevel.Trace,
```

Comment:
Info by default?

Reply:
Since this is a caching focused sample, I think Trace would be better leave as default -we would have to ask customers to switch to it when asking them to provide MSAL logs for debugging later on

Reply:
The concern here is that ppl just copy the sample and go with it. Trace level logging can have a perf impact.